tests/test_utils.py

```diff
@@ -615,10 +615,15 @@ def test_callback_config_file_defaults(pyproject_param, new_default, make_config
     # Create a "compile" run example pointing to the config file
     ctx = Context(compile_cli)
     ctx.params["src_files"] = (str(config_file),)
+    cli_opts = {
+        opt: option
+        for option in ctx.command.params
+        for opt in itertools.chain(option.opts, option.secondary_opts)
+    }
```

Comment:
To DRY this out, could we extract this to a function, like 'get_cli_options(ctx)'?